kernels/x86.py

```diff
@@ -45,17 +45,18 @@
 def is_imm(line): return '$' in line
 def is_memory(line): return '(' in line and 'lea' not in line and 'nop' not in line
 def is_mem_imm(line): return is_memory(line) and is_imm(line)
-def is_mem_load(line): return is_memory(line) and (is_type(LOAD_ANY, line) or C.any_in(('broadcast', 'insert', 'gather'), line))
-def is_mem_store(line): return is_memory(line) and is_type(STORE, line) and C.any_in(('mov', EXTRACT, 'scatter'), line)
+def is_mem_load(line): return is_memory(line) and (is_type(LOAD_ANY, line) or C.any_in(('broadcast', 'insert', 'gather'), line)) 
+def is_mem_store(line): return is_memory(line) and is_type(STORE, line) or C.any_in((EXTRACT, 'scatter'), line)
```

Comment:
Yes its better to separate the 2 checks as this will remove the new code in get_mem_inst() below, consider adding a new is_mem_RMW() and restoring the original code in get_mem_inst(). As Ahmad said MRN code will then call both functions.